context-propagation/src/main/java/io/micrometer/context/logging/slf4j/SelectiveMdcThreadLocalAccessor.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.context.logging.slf4j;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.micrometer.context.ThreadLocalAccessor;
+import org.slf4j.MDC;
+
+/**
+ * {@link ThreadLocalAccessor} for copying values for selected keys from the {@link MDC}
+ * across {@link Thread} boundaries. Ensure no other {@link ThreadLocalAccessor} interacts
+ * with any MDC key that is used, for instance by tracing libraries.
+ *
+ * @author Dariusz Jędrzejczyk
+ * @since 1.1.1
+ */
+public class SelectiveMdcThreadLocalAccessor implements ThreadLocalAccessor<Map<String, String>> {
```

Comment:
Couldn't we unify the two into one implementation, so it can be configured with a list of keys, or with none in which case it gets the entire content? A single `Slf4jThreadLocalAccessor`, possibly with static factory methods names appropriately for each scenario.

Reply:
A drawback of creating a single accessor would be an if statement in all the implemented methods that drives the behaviour. While the JIT would easily optimize that, the code would be less elegant. Let me know if you're ok with that.

Reply:
As discussed separately, I refactored the implementation to include two internal delegate classes to handle a particular case. I skipped factories though, it seems the constructor can potentially suffice?